Apps/phredmine/redmine_connector.py

```diff
@@ -0,0 +1,996 @@
+            max_tickets = param[phantom.APP_JSON_CONTAINER_COUNT]
+            ret_val, max_tickets = self._validate_integer(action_result, max_tickets, REDMINE_CONTAINER_COUNT_KEY)
+            if phantom.is_fail(ret_val):
+                return action_result.get_status()
+            last_run = backfill
+        else:
+            if not last_run:
+                self.debug_print("Run Mode: First Scheduled Poll")
+                last_run = backfill
+            else:
+                self.debug_print("Run Mode: Scheduled Poll")
+                last_run = dateutil.parser.isoparse(last_run)
+
+        self.debug_print(f"Last Run: {last_run}")
+
+        tickets = []
+        try:
+            ret_val, tickets = self._retrieve_tickets(
+                action_result, last_run, max_tickets
+            )
+            if phantom.is_fail(ret_val):
+                return action_result.get_status()
+        except Exception as e:
+            error_msg = self._get_error_message_from_exception(e)
+            action_result.set_status(
+                phantom.APP_ERROR, REDMINE_ERR_FETCHING_TICKETS.format(error=error_msg)
+            )
+
+        self.debug_print(f"Total tickets fetched: {len(tickets)}")
+        self.save_progress(f"Total tickets fetched: {len(tickets)}")
+        for ticket in tickets:
+            self._save_ticket_container(action_result, ticket)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_create_ticket(self, param):
+        """Creates a new ticket on Redmine"""
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier())
+        )
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # Parameters
+        subject = param["subject"]
+        description = param["description"]
+        priority = param.get("priority")
+        tracker = param.get("tracker")
+        custom_fields = param.get("custom_fields", "{}")
+
+        payload = {
+            "issue": {
+                "project_id": self._project_id,
+                "subject": subject,
+                "description": description,
+            }
+        }
+
+        try:
+            parsed_custom_fields = json.loads(custom_fields)
+        except Exception as e:
+            error_msg = self._get_error_message_from_exception(e)
+            return action_result.set_status(
+                phantom.APP_ERROR, REDMINE_ERR_PARSING_CUSTOM_FIELDS.format(error=error_msg)
+            )
+
+        payload["issue"]["custom_fields"] = parsed_custom_fields
+
+        if priority:
+            payload["issue"]["priority_id"] = self._retrieve_enumeration_id(
+                action_result,
+                priority,
+                "issue_priorities",
+                "/enumerations/issue_priorities.json",
+            )
+
+        if tracker:
+            payload["issue"]["tracker_id"] = self._retrieve_enumeration_id(
```

Comment:
Here what if retrieving enumeration id fails for given enum string, should we ignore the tracker/priority by adding proper debug print or have to fail the action? Else we can add the logic as per the above comment for priority.

Reply:
@dfederschmidt, should we ignore or fail in this case?

Reply:
@arathore-crest I think we should fail in the case that `retrieve_enumeration_id` returns an error, good catch.

Reply:
Updated